src/main/java/org/commcare/modern/engine/cases/RecordObjectCache.java

```diff
@@ -0,0 +1,32 @@
+package org.commcare.modern.engine.cases;
+
+import org.commcare.cases.model.Case;
+import org.commcare.cases.query.QueryCache;
+
+import java.util.HashMap;
+
+/**
+ * A straightforward cache object query cache. Stores objects by their record ID.
+ *
+ * Used by other optimizations to isolate doing bulk loads and ensure that they are relevant
+ * when they occur
+ *
+ * Created by ctsims on 6/22/2017.
+ */
+
+public class RecordObjectCache<T> implements QueryCache {
```

Comment:
They get written the bulk storage layer when we do a bulk read

Basically this is a cache lifecycle object where we store something like 

Case ID 2 -> [fully loaded case object from storage]

[Note: This is just the cacheobjectcasche from before, just namechanged to be more generic]

We use the other cache (the record set cache) to keep around lists of cases which should be bulk loaded together, but that list persists across diferent events, we separate them so this cache can be cleared at the end of various queries.